include/interfaces/template.h

```diff
@@ -0,0 +1,152 @@
+#ifndef _pidoums_template_h_
+#define _pidoums_template_h_
+
+#include "interface.h"
+
+template <int dim, int spacedim>
+class ProblemTemplate : public Interface<dim,spacedim,/*n_components=*/dim, ProblemTemplate<dim,spacedim> /*LAC=LATrilinos*/ >
+{
+public:
+  ~ProblemTemplate () {};
+  ProblemTemplate ();
+
+  void declare_parameters (ParameterHandler &prm);
+  void parse_parameters_call_back ();
+
+  // interface with the Interface :)
+
+
+  // this function allows to define the update_flags.
+  // by default they are
+  //  (update_quadrature_points |
+  //  update_JxW_values |
+  //  update_values |
+  //  update_gradients)
+  //
+  //  UpdateFlags get_matrices_update_flags() const;
+
+  // this function allows to set particular update_flags on the face
+  // by default it returns
+  // (update_values         | update_quadrature_points  |
+  //  update_normal_vectors | update_JxW_values);
+  //
+  // UpdateFlags get_face_update_flags() const;
+
+  // this function defines the order of the mapping used when
+  // Dirichlet boundary conditions are applied, when the Initial
+  // solution is interpolated, when the solution vector is stored in
+  // vtu format and when the the error_from_exact is performed.
+  // By default it returns 1;
+  //
+  // unsigned int set_order_of_mapping () const;
+
+  // set the number of matrices to be assembled
+  unsigned int get_number_of_matrices() const
+  {
+    return 2;
+  }
+
+  // Coupling between the blocks of the finite elements in the system:
+  //  0: No coupling
+  //  1: Full coupling
+  //  2: Coupling only on faces
+  // If your matrices are fully coupled (as in this case), you can skip the
+  // implementation because it is already set by default.
+  //
+  // void set_matrices_coupling (std::vector<std::string> > &couplings) const
+  // {
+  //   couplings[0]="1";
+  //   couplings[1]="1";
+  // }
+
+
+  template <typename EnergyType, typename ResidualType>
+  void set_energies_and_residuals(const typename DoFHandler<dim,spacedim>::active_cell_iterator &cell,
```

Comment:
again... set or get? :) 

`set_xxx`: call this function to set an internal variable to the given argument
`get_xxx`: call this function to get the value of an internal variable
`compute_xxx`/`assemble_xxx`: call this function to perform computations